core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -289,6 +345,15 @@ public void addAdditionalRequestHeaders(Iterable<? extends Entry<? extends CharS
         additionalRequestHeaders = createAdditionalHeadersIfAbsent().toBuilder().addObject(headers).build();
     }
 
+    private HttpHeaders createAdditionalHeadersIfAbsent() {
```

Comment:
Unrelated to this PR, but is this needed? I think since we have immutable headers now, we should be able to initialize the field to empty instead of null without overhead.

Reply:
Indeed. Updated them all. :+1: